Builds/CMake/CMakeFuncs.cmake

```diff
@@ -594,27 +608,27 @@ macro(setup_build_boilerplate)
     endif (is_gcc)
   else(NOT WIN32)
     add_compile_options(
-      /bigobj              # Increase object file max size
-      /EHa                 # ExceptionHandling all
-      /fp:precise          # Floating point behavior
-      /Gd                  # __cdecl calling convention
-      /Gm-                 # Minimal rebuild: disabled
-      /GR                  # Enable RTTI
-      /Gy-                 # Function level linking: disabled
+      /bigobj            # Increase object file max size
+      /EHa               # ExceptionHandling all
+      /fp:precise        # Floating point behavior
+      /Gd                # __cdecl calling convention
+      /Gm-               # Minimal rebuild: disabled
+      /GR                # Enable RTTI
+      /Gy-               # Function level linking: disabled
       /FS
-      /MP                  # Multiprocessor compilation
-      /openmp-             # pragma omp: disabled
-      /Zc:forScope         # Language extension: for scope
-      /Zi                  # Generate complete debug info
-      /errorReport:none    # No error reporting to Internet
-      /nologo              # Suppress login banner
-      /W3                  # Warning level 3
-      /WX-                 # Disable warnings as errors
-      /wd"4018"
-      /wd"4244"
-      /wd"4267"
-      /wd"4800"            # Disable C4800(int to bool performance)
-      /wd"4503"            # Decorated name length exceeded, name was truncated
+      /MP                # Multiprocessor compilation
+      /openmp-           # pragma omp: disabled
+      /Zc:forScope       # Language extension: for scope
```

Comment:
Nit: `Language extension: for scope` makes it sounds like this option is enabling a language extension instead of forcing language _conformance_. Reword to `Language conformance: for scope`? 

Also, according to this: https://msdn.microsoft.com/en-us/library/84wcsx8x.aspx, this option is on by default since VS 2015, so we may not need this anymore. (I would _not_ remove the option in this changeset; cleaning up the VS options should be it's own PR).